apps/mobile/src/components/TradeChecklistFlow/components/CalculateAmountFlow/components/CalculateAmountScreen/components/PremiumOrDiscount.tsx

```diff
@@ -0,0 +1,70 @@
+import {getTokens, Stack, Text, XStack} from 'tamagui'
+import {useAtom, useAtomValue, useSetAtom} from 'jotai'
+import {
+  applyFeeOnFeeChangeActionAtom,
+  feeAmountAtom,
+  premiumOrDiscountEnabledAtom,
+} from '../../../atoms'
+import chevronRightSvg from '../../../../../../../images/chevronRightSvg'
+import Image from '../../../../../../Image'
+import {TouchableOpacity} from 'react-native'
+import Switch from '../../../../../../Switch'
+import {useTranslation} from '../../../../../../../utils/localization/I18nProvider'
+import {type NavigationProp, useNavigation} from '@react-navigation/native'
+import {type TradeChecklistStackParamsList} from '../../../../../../../navigationTypes'
+
+function PremiumOrDiscount(): JSX.Element {
+  const {t} = useTranslation()
+  const navigation: NavigationProp<TradeChecklistStackParamsList> =
+    useNavigation()
+  const [premiumOrDiscountEnabled, setPremiumOrDiscountEnabled] = useAtom(
+    premiumOrDiscountEnabledAtom
+  )
+  const feeAmount = useAtomValue(feeAmountAtom)
+  const applyFeeOnFeeChange = useSetAtom(applyFeeOnFeeChangeActionAtom)
+
+  return (
+    <Stack>
+      <XStack ai={'center'} jc={'space-between'} mb={'$4'}>
+        <Text fos={20} ff={'$body600'} col={'$white'}>{`% ${t(
+          'tradeChecklist.calculateAmount.premiumOrDiscount'
+        )}`}</Text>
+        <Switch
+          value={premiumOrDiscountEnabled}
+          onChange={() => {
+            setPremiumOrDiscountEnabled(!premiumOrDiscountEnabled)
+            applyFeeOnFeeChange(0)
```

Comment:
Might be a nice touch to do this: 

```suggestion
            applyFeeOnFeeChange(premiumOrDiscountEnabled ? 0  : feeAmount )
```